src/cli/eest/make/commands/test.py

```diff
@@ -65,47 +65,73 @@ def test():
 
-    file_name = f"test_{test_name}.py"
+    existing_dirs = [d.name for d in base_path.iterdir() if d.is_dir() and d.name != "__pycache__"]
 
-    directory_path = Path("tests") / fork.lower() / f"eip{eip_number}_{test_name}"
-
-    file_path = directory_path / file_name
+    location_choice = input_select(
+        "Select test directory",
+        choices=[
+            {"name": "Use current location", "value": "current"},
+            *existing_dirs,
+            {"name": "** Create new sub-directory **", "value": "new"},
+        ],
+    )
 
-    if file_path.exists():
+    if location_choice == "new":
+        eip_number = input_text("Enter the EIP number").strip()
+        eip_name = input_text("Enter the EIP name").strip()
+        directory_name = input_text("Enter directory name (part after eipXXXX_)").strip()
+        dir_name = f"eip{eip_number}_{directory_name}"
+        directory_path = base_path / dir_name
+        raw_module = input_text("Enter module name (snake_case)").strip()
+        module_name = raw_module if raw_module.startswith("test_") else f"test_{raw_module}"
+    elif location_choice == "current":
+        eip_number = input_text("Enter the EIP number").strip()
+        eip_name = input_text("Enter the EIP name").strip()
+        raw_module = input_text("Enter module name (snake_case)").strip()
+        module_name = raw_module if raw_module.startswith("test_") else f"test_{raw_module}"
+        directory_path = base_path
+    else:
+        dir_parts = location_choice.split("_")
+        eip_number = dir_parts[0][3:]
+        eip_name = " ".join(dir_parts[1:]).title()
+        raw_module = input_text("Enter module name (snake_case)").strip()
+        module_name = raw_module if raw_module.startswith("test_") else f"test_{raw_module}"
+        directory_path = base_path / location_choice
+
+    file_name = f"{module_name}.py"
+    module_path = directory_path / file_name
+
+    if module_path.exists():
         click.echo(
-            click.style(f"\n 🛑 The target test module {file_path} already exists!", fg="red"),
+            click.style(
+                f"\n 🛑 The target test module {module_path} already exists!",
+                fg="red",
+            ),
             err=True,
         )
         sys.exit(1)
 
-    # Create directories if they don't exist
     os.makedirs(directory_path, exist_ok=True)
 
     template = template_env.get_template(f"{test_type.lower()}_test.py.j2")
     rendered_template = template.render(
         fork=fork,
         eip_number=eip_number,
         eip_name=eip_name,
-        test_name=test_name,
+        module_name=module_name,
     )
 
-    with open(file_path, "w") as file:
+    with open(module_path, "w") as file:
         file.write(rendered_template)
 
     click.echo(
         click.style(
-            f"\n 🎉 Success! Test file created at: {file_path}",
+            f"\n �� Success! Test file created at: {module_path}",
```

Comment:
Something weird happened to our party sign :laughing: 
```suggestion
            f"\n 🎉 Success! Test file created at: {module_path}",
```

Reply:
weird :)